src/hotspot/share/oops/klass.cpp

```diff
@@ -260,7 +281,8 @@ Klass::Klass() : _kind(UnknownKlassKind) {
 // The constructor is also used from CppVtableCloner,
 // which doesn't zero out the memory before calling the constructor.
 Klass::Klass(KlassKind kind) : _kind(kind),
-                           _shared_class_path_index(-1) {
+                               _bitmap(SECONDARY_SUPERS_BITMAP_FULL),
```

Comment:
Another observation while browsing the code: `_secondary_supers_bitmap` would be a better name. (Same considerations apply to `_hash_slot`.)

Reply:
This is because the C++ runtime does secondary super cache lookups even before the bitmap has been calculated and the hash table sorted. In this case the bitmap is zero, so teh search thinks there are no secondary supers. Setting _bitmap to SECONDARY_SUPERS_BITMAP_FULL forces a linear search.

I guess there might be a better way to do this. Perhaps a comment is needed?

I agree about `_secondary_supers_bitmap` name.

Reply:
Now it starts to sound concerning... `Klass::set_secondary_supers()` initializes both `_secondary_supers` and `_bitmap` which implies that `Klass::is_subtype_of()` may be called on not yet initialized Klass. It that's the case, it does look like a bug on its own. How is it expected to work when `_secondary_supers` hasn't been set yet?

Reply:
On a second thought the following setter may be the culprit:
```
void Klass::set_secondary_supers(Array<Klass*>* secondaries) {
  assert(!UseSecondarySupersTable || secondaries == nullptr, "");
  set_secondary_supers(secondaries, SECONDARY_SUPERS_BITMAP_EMPTY);
}
```
It should be adjusted to set `SECONDARY_SUPERS_BITMAP_FULL` instead.

Reply:
I've spent a while trying to reproduce the problem but I can't. 

I was seeing a problem where `Klass::is_subtype_of(vmClasses::Cloneable_klass())` was being called before the bitmap had been set. I'm not sure what to think, really. Maybe I should just back out this change to see what happens.

Reply:
I'm in favor of backing out this change and adding an assert/guarantee (on `_secondary_supers != nullptr`) in `Klass::is_subtype_of()` to ensure no subtype checks happen on uninitialized Klasses. Then we should be able to spot and fix all the places where problematic checks happen.